gen/build_deploy/powershell/dcos_node_upgrade.ps1.in

```diff
@@ -0,0 +1,211 @@
+    [Parameter(Mandatory=$false)] [string] $var_dir = 'C:\d2iq\dcos\var'
+)
+
+# GLOBAL
+$global:basedir = "$($install_dir)"
+$global:vardir  = "$($var_dir)"
+
+$ErrorActionPreference = "Stop"
+
+function Write-Log
+{
+    [CmdletBinding()]
+    Param
+    (
+        [Parameter(Mandatory=$true,
+                   ValueFromPipelineByPropertyName=$true)]
+        [ValidateNotNullOrEmpty()]
+        [Alias("LogContent")]
+        [string]$Message,
+
+        [Parameter(Mandatory=$false)]
+        [Alias('LogPath')]
+        [string]$Path="$($vardir)\log\dcos_node_upgrade.log",
+
+        [Parameter(Mandatory=$false)]
+        [ValidateSet("Error","Warn","Info","Debug")]
+        [string]$Level="Info",
+
+        [Parameter(Mandatory=$false)]
+        [switch]$NoClobber
+    )
+
+    Begin
+    {
+        # Set VerbosePreference to Continue so that verbose messages are displayed.
+        $VerbosePreference = 'Continue'
+    }
+    Process
+    {
+
+        # If the file already exists and NoClobber was specified, do not write to the log.
+        if ((Test-Path $Path) -AND $NoClobber) {
+            Write-Error "Log file $Path already exists, and you specified NoClobber. Either delete the file or specify a different name."
+            Return
+            }
+
+        # If attempting to write to a log file in a folder/path that doesn't exist create the file including the path.
+        elseif (!(Test-Path $Path)) {
+            Write-Verbose "Creating $Path."
+            $NewLogFile = New-Item $Path -Force -ItemType File
+            }
+
+        else {
+            # Nothing to see here yet.
+            }
+
+        # Format Date for our Log File
+        $FormattedDate = Get-Date -Format "yyyy-MM-dd HH:mm:ss"
+
+        # Write message to error, warning, or verbose pipeline and specify $LevelText
+        switch ($Level) {
+            'Error' {
+                Write-Error $Message
+                $LevelText = 'ERROR:'
+                }
+            'Warn' {
+                Write-Warning $Message
+                $LevelText = 'WARNING:'
+                }
+            'Info' {
+                Write-Host $Message
+                $LevelText = 'INFO:'
+                }
+            'Debug' {
+                Write-Verbose $Message
+                $LevelText = 'DEBUG:'
+                }
+            }
+
+        # Write log entry to $Path
+        "$FormattedDate $LevelText $Message" | Tee-Object -FilePath $Path -Append
+    }
+    End
+    {
+    }
+}
+
+function Test-CalledFromPrompt {
+    (Get-PSCallStack)[-2].Command -eq "prompt"
+}
+
+function Invoke-NativeApplication {
+    param
+    (
+        [ScriptBlock] $ScriptBlock,
+        [int[]] $AllowedExitCodes = @(0),
+        [switch] $IgnoreExitCode
+    )
+    [string] $stringScriptBlock = $ScriptBlock.ToString();
+    $backupErrorActionPreference = $ErrorActionPreference;
+    $ErrorActionPreference = "Continue";
+    try
+    {
+        if (Test-CalledFromPrompt)
```

Comment:
What am I missing? The two branches here appear to be the same.

Reply:
Fixed by removing stderr to stdout output if a command run in a subcmd (prompt), as it's redundant append. Tested